tools/changelog.js

```diff
@@ -0,0 +1,54 @@
+const {Transform} = require("stream");
+
+var entry;
+
+function nextLine(t) {
+	var offset = t.buffer.indexOf("\n");
+	if (offset >= 0) {
+		var o = offset;
+		if (t.buffer[o - 1] === 13) o--;  // drop carriage-return before line-feed
+		t.emit("line", t.buffer.toString("utf8", 0, o));
+		t.buffer = t.buffer.slice(offset + 1);
+		process.nextTick(nextLine, t);
+	} else
+		t.callback();
+}
+
+process.stdin.pipe(
+	new Transform({
+		transform(chunk, encoding, callback) {
+			this.buffer = this.buffer ?
+				Buffer.concat([this.buffer, chunk], this.buffer.length + chunk.length) :
+				Buffer.from(chunk);
+			this.callback = callback;
+			nextLine(this);
+		},
+		flush(callback) {
+			if (this.buffer && this.buffer.length > 0)  // add line-feed to final line
+				this._transform(Buffer.from("\n"), undefined, callback);
+			else
+				callback();
+		}
+	})
+	.on("line", function(line) {
+		if (line.startsWith("commit")) {
+			if (entry && entry.length > 2) this.push(entry.join(""));
+			entry = undefined;
+		} else if (/^Date:\s+(.*)$/.test(line))
+			entry = ["# " + RegExp.$1 + "\n"];
+		else if (entry && /^\s+\S/.test(line)) {
+			if (entry.length === 1)
+				entry.push("");
```

Comment:
Will that ignore the first line with non-whitespace text? Why?

A test case would be useful: move script to `lib` and add a mocha test file in `test`.

Reply:
The first line contains the name and number of the pull request. Is that useful in the changelog?

Reply:
No. A mocha test for `changelog.js` would really be helpful.